crates/swc_ecma_minifier/src/compress/optimize/strings.rs

```diff
@@ -137,4 +137,31 @@ impl Optimizer<'_> {
             _ => {}
         }
     }
+
+    /// Convert string literals with escaped newline `'\n'` to template literal
+    /// with newline character.
+    pub(super) fn reduce_escaped_newline_for_str_lit(&mut self, expr: &mut Expr) {
+        if self.options.ecma < EsVersion::Es2015
+            || !self.options.experimental.reduce_escaped_newline
+        {
+            return;
+        }
+
+        if let Expr::Lit(Lit::Str(s)) = expr {
+            if s.value.contains('\n') {
+                *expr = Expr::Tpl(Tpl {
+                    span: s.span,
+                    exprs: Default::default(),
+                    quasis: vec![TplElement {
+                        span: s.span,
+                        cooked: Some(s.value.clone()),
+                        raw: s.value.clone(),
```

Comment:
This may cause an issue.

Reply:
The function for `Tpl#raw` => `Str#value` is at https://rustdoc.swc.rs/swc_core/ecma/utils/swc_ecma_ast/struct.Str.html#method.from_tpl_raw , but let me search for `Str` => `Tpl` function...

Reply:
You should use functions at https://github.com/swc-project/swc/blob/f33b0bc2d1474232f608847a6a26eaa068f2c106/crates/swc_ecma_minifier/src/compress/pure/strings.rs#L573-L593

Reply:
This function replace `'\n'` to `"\\n"` for `Tpl#raw`, but I just want to preserve the `'\n'`. It seems little bit redundant if replace `"\\n"` back to `'\n'`.

Reply:
If so, can you add test cases for inputs that contain those strings, respectively?

One test per the search text in "", like one for "\\$" in `convert_str_value_to_tpl_cooked` (you need 3 test cases for the function) and   and "$" for `convert_str_value_to_tpl_raw`?

Reply:
`convert_str_value_to_tpl_cooked` will replace `"\\\\"` to `"\\"`. This behavior results in `Tpl#cooked` missing a `'\\'` character, is it expected?

Reply:
```js
"\\\\"
```

is _not_ identical to

```js
`\\`
```


Oh? I think it's a bug of the function. Could you fix it?

Reply:
Maybe I need to search for the usage of the function because it can be a function with the wrong name. (e.g., Minifier is invoking the function with `Str#raw`, just if it's the case)

Reply:
In `pure/strings.rs` and `pure/misc.rs`, it appears to be invoked with `Str#value`...

Reply:
I use a new fn to convert `Str#value` to `Tpl#raw` without replacing for `\n` and test for some edge cases.

Reply:
Does the original function need to be fixed in this PR?

Reply:
It’s fine to skip for this PR

Reply:
Any other suggestions for this review(or above)?